### PR TITLE
Name Transformation

### PR DESCRIPTION
**Pull request by parisholley**
Closed #3498 in favor of this 2.0 merged PR.

Supports #3497

### CODE REVIEW

**Comment by parisholley** [comment]
@leezen @infin8x i've had this PR effectively open since 2019, when can we get this included?

**Comment by pgavlin** [comment]
@lukehoban @mikhailshilkov can you take another look w.r.t the comments you left on @parisholley's original PR?

File: sdk/dotnet/Pulumi/PublicAPI.Shipped.txt [hunk 1]
```diff
@@ -178,11 +178,9 @@ Pulumi.ResourceTransformationArgs
 Pulumi.ResourceTransformationArgs.Args.get -> Pulumi.ResourceArgs
 Pulumi.ResourceTransformationArgs.Options.get -> Pulumi.ResourceOptions
 Pulumi.ResourceTransformationArgs.Resource.get -> Pulumi.Resource
-Pulumi.ResourceTransformationArgs.ResourceTransformationArgs(Pulumi.Resource resource, Pulumi.ResourceArgs args, Pulumi.ResourceOptions options) -> void
```
Comment (mikhailshilkov) [inline sdk/dotnet/Pulumi/PublicAPI.Shipped.txt:181]: We can't take this breaking change in a minor version. Could we provide a new constructor overload instead?

**Comment by ESipalis** [comment]
Would be nice to get this in, so we don't have to manually prefix parent resource name for every resource.

**Comment by mikhailshilkov** [comment]
As this PR is multiple years old now, I'll have to close it. Kudos to @parisholley for this initial version, too bad we haven't brought it over the line. The PR is referenced from the issue, so once we are ready we may re-apply the changes on top of the current code base and find ways to address the existing feedback.

**Pull request closed without merging**